src/java/src/main/java/com/dnanexus/DXWorkflow.java

```diff
@@ -239,4 +242,42 @@ public Describe getCachedDescribe() {
         return ExecutableRunner.getWorkflowRunnerWithEnvironment(this.getId(), this.env);
     }
 
+
+    @JsonInclude(Include.NON_NULL)
+    private static class WorkflowAddStageInput {
+        @JsonProperty
+        public int editVersion;
+
+        @JsonProperty
+        public String name;
+
+        @JsonProperty
+        private JsonNode input;
+
+        @JsonProperty
+        public String executable;
+    }
+
+    @JsonIgnoreProperties(ignoreUnknown = true)
+    private static class WorkflowAddStageOutput {
+        @JsonProperty
+        public int editVersion;
+
+        @JsonProperty
+        public String stage;
+    }
+
+    public DXStage addStage(DXApplet applet,
+                            String name,
+                            Object stageInputs,
+                            int editVersion) {
```

Comment:
The workflow should probably have a method to retrieve its own editVersion number, since if you implement any other methods on workflows, their callers will also need to supply editVersion. In that case DXStage doesn't need its own editVersion field.

Do you think this class should manage the editVersions itself (i.e. remember the last editVersion of each operation, and supply it on the next operation)? Or should that responsibility be passed on to the user (as you've done so far)?